presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -629,6 +632,26 @@ public static Slice stEnvelope(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
         return serialize(envelope);
     }
 
+    @Description("Returns the lower left and upper right corners of bounding rectangular polygon of a Geometry")
+    @ScalarFunction("ST_EnvelopeAsPts")
+    @SqlType("array(" + GEOMETRY_TYPE_NAME + ")")
+    public static Block stEnvelopeAsPts(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
+    {
+        Envelope envelope = deserializeEnvelope(input);
+        BlockBuilder blockBuilder = VARCHAR.createBlockBuilder(null, 2);
+        if (envelope == null) {
+            VARCHAR.writeSlice(blockBuilder, EMPTY_MULTIPOINT);
+            VARCHAR.writeSlice(blockBuilder, EMPTY_MULTIPOINT);
+        }
+        else {
+            Point point1 = new Point(envelope.getXMin(), envelope.getYMin());
```

Comment:
Perhaps, for clarity, rename `point1` and `point2` to `lowerLeftCorner` and `upperRightCorner`.

Reply:
Done